vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/GroupingTest.java

```diff
@@ -0,0 +1,327 @@
+package com.vaadin.flow.component.spreadsheet.tests;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import com.vaadin.flow.component.spreadsheet.Spreadsheet;
+
+public class GroupingTest {
+
+    private Spreadsheet spreadsheet;
+
+    @Before
+    public void init() {
+        spreadsheet = TestHelper.createSpreadsheet("Groupingtest.xlsx");
+    }
+
+    @Test
+    public void expandRow_alreadyExpanded_shouldNotThrow() {
+        setActiveSheet(SHEET9);
+
+        expandRow(SHEET9_ROW_GROUP);
+    }
+
+    @Test
+    public void collapseRow_rowHidden() {
+        setActiveSheet(SHEET9);
+
+        collapseRow(SHEET9_ROW_GROUP);
+        Assert.assertTrue(spreadsheet.isRowHidden(SHEET9_ROW));
+    }
+
+    @Test
+    public void collapseRow_expandRow_rowVisible() {
+        setActiveSheet(SHEET9);
+
+        collapseRow(SHEET9_ROW_GROUP);
+        expandRow(SHEET9_ROW_GROUP);
+        Assert.assertFalse(spreadsheet.isRowHidden(SHEET9_ROW));
+    }
+
+    @Test
+    public void collapseRow_collapseParentRow_expandRow_rowHidden() {
+        setActiveSheet(SHEET4);
+        // Collapse child row group
+        collapseRow(SHEET4_ROW_GROUP);
+        Assert.assertTrue(spreadsheet.isRowHidden(SHEET4_ROW));
+
+        // Collapse parent row group
+        collapseRow(SHEET4_PARENT_ROW_GROUP);
+
+        // Expand child row group
+        expandRow(SHEET4_ROW_GROUP);
+        Assert.assertTrue(spreadsheet.isRowHidden(SHEET4_ROW));
+    }
+
+    @Test
+    public void collapseRow_collapseParentRow_expandParentRow_rowHidden() {
+        setActiveSheet(SHEET4);
+        // Collapse child row group
+        collapseRow(SHEET4_ROW_GROUP);
+        Assert.assertTrue(spreadsheet.isRowHidden(SHEET4_ROW));
+
+        // Collapse parent row group
+        collapseRow(SHEET4_PARENT_ROW_GROUP);
+
+        // Expand parent row group
+        expandRow(SHEET4_PARENT_ROW_GROUP);
+        Assert.assertTrue(spreadsheet.isRowHidden(SHEET4_ROW));
+    }
+
+    @Test
+    public void collapseParentRow_expandParentRow_rowVisible() {
+        setActiveSheet(SHEET4);
+        // Collapse parent row group
+        collapseRow(SHEET4_PARENT_ROW_GROUP);
+        Assert.assertTrue(spreadsheet.isRowHidden(SHEET4_ROW));
+
+        // Expand Parent row group
+        expandRow(SHEET4_PARENT_ROW_GROUP);
+        Assert.assertFalse(spreadsheet.isRowHidden(SHEET4_ROW));
+    }
+
+    @Test
+    public void inverted_collapseRow_rowHidden() {
```

Comment:
Just a clarification: what does inverted here refer to?

Reply:
Inverted groups is a feature in spreadsheet

regular groups:
<img width="595" alt="Screenshot 2022-10-14 at 17 16 48" src="https://user-images.githubusercontent.com/1222264/195869345-c373f8a8-97ed-42a8-a2c7-4b51e06370f2.png">

inverted groups:
<img width="530" alt="Screenshot 2022-10-14 at 17 16 53" src="https://user-images.githubusercontent.com/1222264/195869367-2bc22506-8b38-4f83-917f-f25d547cdca2.png">

Reply:
Well, probably doesn't go by that name officially, but that naming was used in the GroupingUtil source code